v2/swagger.yaml

```diff
@@ -0,0 +1,629 @@
+        - in: path
+          name: name
+          type: string
+          required: true
+          description: component name
+        - in: path
+          name: ver
+          type: string
+          required: true
+          description: component version
+      responses:
+        '200':
+          schema:
+            $ref: "#/definitions/ComponentAnalysesResponse"
+          description: Detailed response for the requested application component
+        '401':
+          description: Request unauthorized
+        '404':
+          description: Data not found
+  '/stack-analyses':
+    post:
+      tags:
+        - Scan Services
+      operationId: f8a_scanner.api_v1.stack-analyses-post
+      summary: Initiates a scan to gather security and other details about an application stack and related recommendations to improve the stack
+      consumes:
+        - multipart/form-data
+      parameters:
+        - name: ecosystem
+          in: header
+          description: The ecosystem of the stack analysis being called.
+          required: true
+          type: string
+        - name: origin
+          in: header
+          description: This is required to give an indication as to from where is the request getting generated. If from VScode extension, the value which we need to send is 'vscode'.
+          required: false
+          type: string
+        - name: show_transitive
+          in: header
+          description: This is required to enable or disable the transitive support. We need to pass the value as 'true' if transitive is needed. Default value is 'false'.
+          required: false
+          type: string
+        - name: manifest[]
+          in: formData
+          description: A manifest file dependencies.txt for maven, npmlist.json for npm & pylist.json for pypi. Commands to generate these files are mentioned in the README.md
+          required: true
+          type: file
+        - name: filePath[]
+          in: formData
+          description: The absolute or relative path where the manifest file is located. This has to be mandatorily provided with the manifest[] parameter and vice versa.
+          required: true
+          type: string
+
+      responses:
+        '200':
+          schema:
+            $ref: "#/definitions/StackAnalysesPostResponse"
+          description: Response with the unique identifier for the stack analysis request
+        '401':
+          description: Request Unauthorized
+        '400':
+          description: Bad request
+  '/stack-analyses/{request-id}':
+    get:
+      tags:
+        - Scan Services
+      operationId: f8a_scanner.api_v1.stack-analyses-get
+      summary: Get security and other details about an application stack and related recommendations to improve the stack
+      parameters:
+        - in: path
+          name: request-id
+          type: string
+          required: true
+          description: stack analysis request id
+      responses:
```

Comment:
can we have clear separation of response models to represent free / registered user personas ? These two different response models can in turn reuse any common fields

Reply:
I will address in next iteration.